File: solutions/elixir/freelancer-rates/2/lib/freelancer_rates.ex
defmodule FreelancerRates do
  def daily_rate(hourly_rate) do
    hourly_rate
    |> Kernel.*(8.0)
  end

  def apply_discount(before_discount, discount) do
    100
    |> Kernel.-(discount)
    |> Kernel./(100)
    |> Kernel.*(before_discount)
  end

  def monthly_rate(hourly_rate, discount) do
    hourly_rate
    |> daily_rate
    |> Kernel.*(22)
    |> apply_discount(discount)
    |> ceil
  end

  def days_in_budget(budget, hourly_rate, discount) do
    daily_discounted_rate =
      hourly_rate
      |> daily_rate
      |> apply_discount(discount)
      
    budget
    |> Kernel./(daily_discounted_rate)
    |> Float.floor(1)
  end
end

File: solutions/cpp/freelancer-rates/2/freelancer_rates.cpp
#include <cmath>

const int BILLABLE_DAYS = 22;
const int WORK_HOUR = 8;

double daily_rate(double hourly_rate) {
    return WORK_HOUR * hourly_rate;
}

double apply_discount(double before_discount, double discount) {
    return before_discount * (1 - discount / 100);
}

int monthly_rate(double hourly_rate, double discount) {
    return ceil(apply_discount(daily_rate(hourly_rate), discount) * BILLABLE_DAYS);
}

int days_in_budget(int budget, double hourly_rate, double discount) {
    return floor(budget / apply_discount(daily_rate(hourly_rate), discount));
}